src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -1,56 +1,65 @@
 package uk.gov.hmcts.reform.sscs.ccd.presubmit.postponementrequest;
 
+import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
-import static org.apache.commons.collections4.CollectionUtils.emptyIfNull;
-import static uk.gov.hmcts.reform.sscs.model.AppConstants.POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS;
 
-import java.time.LocalDateTime;
-import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
 import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
 import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
-import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
 import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
 
 
 @Service
 public class PostponementRequestAboutToStartHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
+    public static final String NOT_LIST_ASSIST_CASE_ERROR = "Postponement requests can only be made for list"
+        + " assist cases";
+
+    @Value("${feature.snl.enabled}")
+    private boolean isScheduleListingEnabled;
+
     @Override
     public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
         requireNonNull(callback, "callback must not be null");
         requireNonNull(callbackType, "callbacktype must not be null");
 
-        return callbackType.equals(CallbackType.ABOUT_TO_START)
-                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
-                && callback.getCaseDetails() != null;
+        return callbackType == CallbackType.ABOUT_TO_START
+            && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+            && isScheduleListingEnabled;
```

Comment:
Why was the getCaseDetails() != null condition removed? Won't it affect the handle method potentially opening to NullPointerException being thrown in some circumstances?

Reply:
`getCaseDetails()` is not needed and is fine if we throw a `NullPointerException` in that case.

As for the `isScheduleListingEnabled`, the whole concept of postponement is not released yet and it will be released as part of S&L, so effectively the handler will start being called only when the feature flag will be enabled.
In essence, the whole handler is tied to the S&L feature being enabled so we can work under that assumption.

Resolving this conversation.